includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -4253,10 +4253,13 @@ public function wc_payments_get_payment_method_by_id( $payment_method_id ) {
 	 * @return string
 	 */
 	public function get_theme_icon() {
-		$upe_appearance_theme = get_transient( self::UPE_APPEARANCE_THEME_TRANSIENT );
-		if ( $upe_appearance_theme ) {
-			return 'night' === $upe_appearance_theme ? $this->payment_method->get_dark_icon() : $this->payment_method->get_icon();
+		$upe_appearance_theme        = get_transient( self::UPE_APPEARANCE_THEME_TRANSIENT );
+		$upe_blocks_appearance_theme = get_transient( self::WC_BLOCKS_UPE_APPEARANCE_THEME_TRANSIENT );
+
+		if ( 'night' === $upe_appearance_theme || 'night' === $upe_blocks_appearance_theme ) {
```

Comment:
Just FYI, this `get_theme_icon()` function originally does not use the blocks theme transient, because this function is only used to decide the payment method icon to display in the shortcode checkout [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L280). The shortcode checkout logo for payment methods relies on the `icon` property of the payment gateway, whereas the logo for blocks checkout is [inserted into the DOM manually](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/blocks/payment-method-label.js#L59-L67).

I think it's unlikely that shortcode and blocks checkout will require opposite themes, but it's possible and would create an unintended side-effect here. This `get_theme_icon()` function can probably be named more appropriately (ma b, I believe) to express its actual purpose, but I'd probably recommend persisting its original logic. 🤷